langchain/src/embeddings/googlevertexai-image.ts

```diff
@@ -0,0 +1,143 @@
+import { Embeddings, EmbeddingsParams } from "./base.js";
+import {
+  GoogleVertexAIBasePrediction,
+  GoogleVertexAIConnectionParams,
+  GoogleVertexAILLMResponse,
+} from "../types/googlevertexai-types.js";
+import { GoogleVertexAIConnection } from "../util/googlevertexai-connection.js";
+import { AsyncCallerCallOptions } from "../util/async_caller.js";
+// import { chunkArray } from "../util/chunk.js";
+
+export interface GoogleVertexAIImageEmbeddingsParams
+  extends EmbeddingsParams,
+    GoogleVertexAIConnectionParams {}
+
+interface GoogleVertexAIImageEmbeddingsOptions extends AsyncCallerCallOptions {}
+
+interface GoogleVertexAIImageEmbeddingsInstance {
+  text?: string;
+  image?: {
+    bytesBase64Encoded: string;
+  };
+}
+
+interface GoogleVertexImageEmbeddingsResults
+  extends GoogleVertexAIBasePrediction {
+  textEmbedding?: number[];
+  imageEmbedding?: number[];
+}
+
+/**
+ * The media should have a text property, an image property, or both.
+ */
+export type GoogleVertexAIMedia =
+  | {
+      text: string;
+      image?: Buffer;
+    }
+  | {
+      text?: string;
+      image: Buffer;
+    };
+
+export type MediaEmbeddings = {
+  text?: number[];
+  image?: number[];
+};
+
+export class GoogleVertexAIImageEmbeddings
```

Comment:
Do we think it's premature to call this `GoogleVertexAIMultimodalEmbeddings`?

Reply:
I think when I started developing it, they were ambiguous about what its name would be. But they seem to have (mostly) settled on using "multimodal". I'll change the name.